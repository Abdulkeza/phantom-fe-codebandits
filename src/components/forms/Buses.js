import React from 'react';
import Button from '../Button.js';
import { useForm } from 'react-hook-form';

const Buses = ({
  formTitle,
  formAction,
  busType,
  company,
  seats,
  plateNumber
}) => {
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm({
    reValidateMode: 'onChange',
    mode: 'onChange'
  });
  const onValid = async (data) => {
    await formAction(data);
  };
  const inputClassStyles = 'rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]';
  return (
    <div className="px-8 py-4 my-auto font-raleway md:py-8">
      <form
        className="py-8 shadow-main px-4 flex flex-col w-full md:max-w-lg md:mx-auto"
        onSubmit={handleSubmit(onValid)}
      >
        <h1 className="font-extrabold text-2xl pb-8">{formTitle}</h1>
        <label htmlFor="busType" className="font-bold mb-2">
          Bus type
        </label>
        <select
          defaultValue={busType}
          name="busType"
          id="busType"
          {...register('busType')}
          className={inputClassStyles}
        >
          <option hidden>Select bus type</option>
          <option value="Coaster">Coaster</option>
          <option value="Minibus">Minibus</option>
          <option value="Big Bus">Big Bus</option>
        </select>
        <p className="text-red-800">
          {errors?.busType && errors.busType.message}
        </p>
        <label htmlFor="busType" className="font-bold mb-2">
          Company
        </label>
        <select
          defaultValue={company}
          name="company"
          id="company"
          className={inputClassStyles}
          {...register('company')}
        >
          <option hidden>Select company</option>
          <option value="KBS">Kigali Bus Service</option>
          <option value="Royal_express">Royal Express</option>
          <option value="Virunga_express">Virunga Express</option>
        </select>
        <label htmlFor="seats" className="font-bold mb-2">
          Seats
        </label>
        <input
          defaultValue={seats}
          type="number"
          placeholder="Enter number of seats"
          className={inputClassStyles}
          name="seats"
          {...register('seats', {
            required: 'Number of seats is required',
            pattern: {
              value: /^\d*$/,
              message: 'Enter a valid number'
            }
          })}
        />
        <p className="text-red-800 mb-2">
          {errors?.seats && errors.seats.message}
        </p>
        <label htmlFor="seats" className="font-bold mb-2">
          Plate number
        </label>
        <input
          defaultValue={plateNumber}
          type="text"
          placeholder="Enter plate number"
          className={inputClassStyles}
          name="plateNumber"
          {...register('plateNumber', {
            required: 'Bus plate number is required',
            pattern: {
              value: /^[A-Za-z0-9]*$/,
              message: 'Enter a valid plate number'
            }
          })}
        />
        <p className="text-red-800 mb-2">
          {errors?.plateNumber && errors.plateNumber.message}
        </p>
        <div className="self-center">
          <Button
            name={formTitle}
            type="submit"
            styles="bg-primary text-sm text-white py-2 px-3 mt-4"
          />
        </div>
      </form>
    </div>
  );
};

Buses.defaultProps = {
  busType: undefined,
  company: undefined,
  seats: undefined,
  plateNumber: undefined
};

export default Buses;
