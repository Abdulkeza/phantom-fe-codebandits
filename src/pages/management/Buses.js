import React from 'react'

const Buses = () => {
  return (
    <div>Buses management</div>
  )
}

export default Buses