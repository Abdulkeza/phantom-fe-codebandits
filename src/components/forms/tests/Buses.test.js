import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Buses from '../Buses.js';

describe('Buses form component test', () => {
  it('it should render Buses form component', () => {
    const busesForm = renderer
      .create(
        <MemoryRouter>
          <Buses />
        </MemoryRouter>
      )
      .toJSON();
    expect(busesForm).toMatchSnapshot();
  });
  it('it should call form action on submit', () => {
    const formActionMock = jest.fn();
    const wrapper = mount(<Buses formAction={formActionMock()} />);
    const busesForm = wrapper.find('form');
    busesForm.simulate('submit',{
      busType:"Coaster",
      seats:"30",
      company:"Virunga"
    }),
    expect(formActionMock).toBeCalledTimes(1);
  });
});
