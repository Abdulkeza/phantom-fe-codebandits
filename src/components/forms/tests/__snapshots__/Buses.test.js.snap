// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buses form component test it should render Buses form component 1`] = `
<div
  className="px-8 py-4 my-auto font-raleway md:py-8"
>
  <form
    className="py-8 shadow-main px-4 flex flex-col w-full md:max-w-lg md:mx-auto"
    onSubmit={[Function]}
  >
    <h1
      className="font-extrabold text-2xl pb-8"
    />
    <label
      className="font-bold mb-2"
      htmlFor="busType"
    >
      Bus type
    </label>
    <select
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      id="busType"
      name="busType"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        hidden={true}
      >
        Select bus type
      </option>
      <option
        value="coaster"
      >
        Coaster
      </option>
      <option
        value="minibus"
      >
        Minibus
      </option>
      <option
        value="bigBus"
      >
        Big Bus
      </option>
    </select>
    <p
      className="text-red-800"
    />
    <label
      className="font-bold mb-2"
      htmlFor="busType"
    >
      Company
    </label>
    <select
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      id="busType"
      name="company"
    >
      <option
        hidden={true}
      >
        Select company
      </option>
      <option
        value="kbs"
      >
        Kigali Bus Service
      </option>
      <option
        value="royal_express"
      >
        Royal Express
      </option>
      <option
        value="virunga_express"
      >
        Virunga Express
      </option>
    </select>
    <label
      className="font-bold mb-2"
      htmlFor="seats"
    >
      Seats
    </label>
    <input
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      name="seats"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Enter number of seats"
      type="number"
    />
    <p
      className="text-red-800 mb-2"
    />
    <label
      className="font-bold mb-2"
      htmlFor="seats"
    >
      Plate number
    </label>
    <input
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      name="plateNumber"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Enter plate number"
      type="text"
    />
    <p
      className="text-red-800 mb-2"
    />
    <div
      className="self-center"
    >
      <div>
        <button
          className="bg-primary text-sm text-white py-2 px-3 mt-4 rounded-xl py-3 text-base md:text-sm lg:text-lg xl:text-lg px-5"
          type="submit"
        />
      </div>
    </div>
  </form>
</div>
`;
